### PR TITLE
Fixed tests, moved CHT to Dependency Injector

### PR DESCRIPTION
**Pull request by noescape00**
1) Inject CHT instead of creating it in CM
2) Create CM only in one place in the tests (moved this to common tests project). Otherwise every time CM's signature is updated we need to fix several different files
3) Fixed some tests

### CODE REVIEW

File: src/Stratis.Bitcoin.FullNode.sln [hunk 1]
```diff
@@ -128,7 +128,7 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Stratis.Bitcoin.Features.Sm
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Stratis.SmartContracts.Core.Validation.Tests", "Stratis.SmartContracts.Core.Validation.Tests\Stratis.SmartContracts.Core.Validation.Tests.csproj", "{247C062D-8A16-4DD8-93B4-1D1C3D49B498}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "RuntimeObserver", "RuntimeObserver\RuntimeObserver.csproj", "{38F49E88-545A-437B-9225-0BC8815B0281}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "RuntimeObserver", "RuntimeObserver\RuntimeObserver.csproj", "{38F49E88-545A-437B-9225-0BC8815B0281}"
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.FullNode.sln]: Why did this change?

File: src/Stratis.Bitcoin.Tests.Common/CMCreator.cs [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+using Stratis.Bitcoin.Features.Consensus.Rules.CommonRules;
+using Stratis.Bitcoin.Interfaces;
+using Stratis.Bitcoin.P2P;
+using Stratis.Bitcoin.P2P.Peer;
+using Stratis.Bitcoin.P2P.Protocol.Payloads;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Tests.Common
+{
+    public static class CMCreator
```
Comment (fassadlr) [inline src/Stratis.Bitcoin.Tests.Common/CMCreator.cs]: Ivan, lets stick to proper naming ;)

Call this `ConsensusManagerHelper` and the method just `Create`.

File: src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs [hunk 3]
```diff
@@ -149,7 +149,7 @@ public interface IChainedHeaderTree
     }
 
     /// <inheritdoc />
-    internal sealed class ChainedHeaderTree : IChainedHeaderTree
+    public class ChainedHeaderTree : IChainedHeaderTree
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs]: We can still seal this as we dont' want it being inheritable at this point :)

File: src/Stratis.Bitcoin/Base/BaseFeature.cs [hunk 4]
```diff
@@ -367,6 +367,7 @@ public static IFullNodeBuilder UseBaseFeature(this IFullNodeBuilder fullNodeBuil
 
                     // Consensus
                     services.AddSingleton<IConsensusManager, ConsensusManager>();
+                    services.AddSingleton<IChainedHeaderTree, ChainedHeaderTree>();
```
Comment (fassadlr) [inline src/Stratis.Bitcoin/Base/BaseFeature.cs]: 💯

File: src/Stratis.Bitcoin.FullNode.sln (hunk 1 above)
Comment (noescape00) [reply src/Stratis.Bitcoin.FullNode.sln]: Hm, didn't notice VS did that. Will revert, this one is not intended.

File: src/Stratis.Bitcoin/Consensus/ConsensusManager.cs [hunk 5]
```diff
@@ -315,6 +314,8 @@ private void PeerDisconnectedLocked(int peerId)
 
             if (removed)
             {
+                // TODO Use node lifetime here and don't call ProcessDownloadQueueLocked and chainedHeaderTree.PeerDisconnected
+                // on node shutdown. Otherwise we have big shutdown perf hit.
```
Comment (Aprogiena) [inline src/Stratis.Bitcoin/Consensus/ConsensusManager.cs]: Can you explain what you're talking about here?

**Pull request merged**